spec/policy_machine_storage_adapters/active_record_spec.rb

```diff
@@ -603,6 +603,109 @@
       end
     end
 
+    describe '#pluck_ancestor_tree' do
+      let(:user_attr_4) { pm1.create_user_attribute('user_attr_4') }
+      let(:user_attr_5) { pm1.create_user_attribute('user_attr_5') }
+      let(:user_attr_6) { pm1.create_user_attribute('user_attr_6') }
+      let!(:single_ancestors) { [user_attr_4, user_attr_5, user_attr_6] }
+
+      let(:user_attr_7) { pm1.create_user_attribute('user_attr_7') }
+      let(:user_attr_8) { pm1.create_user_attribute('user_attr_8') }
+      let(:user_attr_9) { pm1.create_user_attribute('user_attr_9') }
+      let!(:double_ancestors) { [user_attr_7, user_attr_8, user_attr_9] }
+
+      let(:user_attr_10) { pm1.create_user_attribute('user_attr_10') }
+
+      before do
+        darken_colors.call
+
+        single_ancestors.each { |ancestor| ancestor.update(color: 'gold' ) }
+        double_ancestors.each { |ancestor| ancestor.update(color: 'silver' ) }
+        pm1.add_assignment(user_attr_4, user_attr_1)
+        pm1.add_assignment(user_attr_5, user_attr_1)
+        pm1.add_assignment(user_attr_6, user_attr_1)
+
+        pm1.add_assignment(user_attr_7, user_attr_4)
+        pm1.add_assignment(user_attr_8, user_attr_5)
+        pm1.add_assignment(user_attr_9, user_attr_6)
+      end
+
+      context 'no filter is applied' do
+        it 'returns appropriate ancestors and the specified attribute' do
+          plucked_results = {
+            'user_1' => [],
```

Comment:
any reason not to use colons instead of hash rockets?

Reply:
In the method you specifically delete the key for the originator's id. Why is it in its own ancestor tree?

Reply:
The key-value pairs are formed from the `child_id`-`parent_id` relationship. `assignments_recursive` starts by specifying the first `child_id` as the root node. Thus it is returned as a key in its own ancestor tree. I manually delete it, since a) it is implied by calling the method on it, and b) it doesn't necessarily satisfy the filters on its ancestors.

Perfect example: `ct.pluck_ancestor_tree(filters: { active_policy_element_type: 'ConfigurationTypeRole' }, fields: [:name, :tags])` The ConfigurationType doesn't satisfy the CTR ape_type filter, so we nix it.